api/src/main/java/org/pmiops/workbench/survey/NewUserSatisfactionSurveyServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package org.pmiops.workbench.survey;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import org.pmiops.workbench.access.AccessTierService;
+import org.pmiops.workbench.db.dao.UserAccessTierDao;
+import org.pmiops.workbench.db.model.DbAccessTier;
+import org.pmiops.workbench.db.model.DbUser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class NewUserSatisfactionSurveyServiceImpl implements NewUserSatisfactionSurveyService {
+
+  private final UserAccessTierDao userAccessTierDao;
+  private AccessTierService accessTierService;
+  private final Clock clock;
+
+  @Autowired
+  public NewUserSatisfactionSurveyServiceImpl(
+      UserAccessTierDao userAccessTierDao, AccessTierService accessTierService, Clock clock) {
+    this.userAccessTierDao = userAccessTierDao;
+    this.accessTierService = accessTierService;
+    this.clock = clock;
+  }
+
+  @Override
+  public boolean eligibleToTakeSurvey(DbUser user) {
+    if (user.getNewUserSatisfactionSurvey() != null) {
+      return false;
+    }
+
+    DbAccessTier registeredAccessTier = accessTierService.getRegisteredTierOrThrow();
+
+    return userAccessTierDao
```

Comment:
this is fine as-is but as an alternative...

since you already have `accessTierService` you could instead use `getAccessTiersForUser()` and filter on registered tier, without needing to also have the Dao here.

shrug.  your choice.

Reply:
`getAccessTiersForUser` returns a `DbAccessTier` and `getByUserAndAccessTier` returns a `DbUserAccessTier`. In this case, I specifically need to know the enabled date for that user so I need a `DbUserAccessTier`. I don't see any `accessTierService` that would return that type. Let me know if that sounds right.

Is there a preference against specifically using DAOs here, or just against having additional dependencies?